app/assets/javascripts/modules/download_all.js

```diff
@@ -0,0 +1,34 @@
+// Module ensures Download All button doesn't get clicked repeatedly The first
+// click on Download button will cause the button text to be changed
+// "Initializing download" and disabled, then the download will start.
+
+(function(global) {
+  var Module = (function() {
+    return {
+
+      init: function() {
+        this.waitForClick();
+      },
+
+      waitForClick: function() {
```

Comment:
Is there a reason this can't be merged into the `init()` function?

Reply:
Not that I'm aware of.